app/src/main/java/com/example/perfectweatherallyear/ui/weekWeather/WeekWeatherFragment.kt

```diff
@@ -10,16 +10,19 @@ import androidx.recyclerview.widget.LinearLayoutManager
 import com.example.perfectweatherallyear.changeFragment
 import com.example.perfectweatherallyear.databinding.FragmentWeekWeatherBinding
 import com.example.perfectweatherallyear.model.DayWeather
+import com.example.perfectweatherallyear.repository.Repository
+import com.example.perfectweatherallyear.repository.remoteData.WeatherData.RemoteDataSource
+import com.example.perfectweatherallyear.ui.ViewModelFactory
 import com.example.perfectweatherallyear.ui.detailWeather.ARG_DAY_WEATHER
 import com.example.perfectweatherallyear.ui.detailWeather.DetailWeatherFragment
 import com.google.gson.GsonBuilder
 
 class WeekWeatherFragment : Fragment() {
-    private var weekWeatherMap: Map<String, DayWeather> = mapOf()
+    private var weekWeatherList: List<DayWeather> = listOf()
     lateinit var weatherForecastAdapter: WeatherForecastAdapter
     private lateinit var binding: FragmentWeekWeatherBinding
 
-    private val weekWeatherViewModel by viewModels<WeekWeatherViewModel>()
+    private val weekWeatherViewModel by viewModels<WeekWeatherViewModel>{ ViewModelFactory(Repository(RemoteDataSource())) }
```

Comment:
Fragment should't create Repository and RemoteDataSource by itself. Implement Separate Factories for this